app/models/primer.rb

```diff
@@ -1,9 +1,13 @@
 class Primer < ActiveRecord::Base
 
   has_many :marker_assays_A, class_name: 'MarkerAssay',
-             foreign_key: 'primer_a'
+             foreign_key: 'primer_a_id'
   has_many :marker_assays_B, class_name: 'MarkerAssay',
-           foreign_key: 'primer_b'
+             foreign_key: 'primer_b_id'
+
+  def marker_assays
+    marker_assays_A | marker_assays_B
```

Comment:
See my response to your previous comment (in `primer.rb`). I can delete it if you wish.